internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -648,16 +678,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		r.record.Event(xr, event.Normal(reasonPublish, "Successfully published connection details"))
 	}
 
-	warnings := 0
-	for _, e := range res.Events {
+	warning := false
+	for _, e := range append(res.CompositeEvents, res.ClaimEvents...) {
 		if e.Type == event.TypeWarning {
-			warnings++
+			warning = true
+			break
 		}
-		log.Debug(e.Message)
-		r.record.Event(xr, e)
```

Comment:
Info: The log and event recording is now being handled by the `handleFunctionResults` logic.